### PR TITLE
Add CI integrations

### PR DESCRIPTION
**Pull request by andystanton**
I added some unit tests in #23. This PR adds a .travis.yml so that the tests can be triggered on PRs and merges.

If other maintainers agree this is a good thing, we need to add a Travis build job for this repository (when this is done, it will show here: https://travis-ci.org/AtomLinter/) before this PR is merged. Any maintainer can do this, but a discussion is probably useful first.

### CODE REVIEW

**Comment by andystanton** [comment]
Example Travis CI build from my fork: https://travis-ci.org/andystanton/linter-scalac/builds/99337143

**Comment by Arcanemagus** [comment]
As you can see from the link, many of the repositories in this org already are building on TravisCI, and on CircleCI and on AppVeyor. The more the better :smile:

I would recommend you look over https://github.com/atom/ci as that is what we use as the base for configuration of builds usually. However when setting up AppVeyor please use something like [this](https://github.com/AtomLinter/linter-bootlint/blob/master/appveyor.yml) as the template, for some reason https://github.com/atom/ci/pull/12 has yet to be merged which means the master branch will install Atom v1.0.7 unless you modify it like above.

Using the `.travis.yml` from that repository will by default have Travis-CI test the stable and beta channels of Atom on both Linux and OS X.

**Comment by Arcanemagus** [comment]
Also just an FYI: I think you are the only active maintainer of this specific project, all recent commits have been by org owners keeping things working :wink:.

**Comment by andystanton** [comment]
Thanks for the reference @Arcanemagus - I'll update this PR with the other CircleCI and AppVeyor and bring the Travis configuration in line.

**Comment by Arcanemagus** [comment]
Oh, a note about CircleCI: For some bizarre reason when they updated their build containers last they _downgraded_ `wget`, you need to put in something like I mention [here](https://github.com/atom/ci/pull/28#issuecomment-163088382) to get it to install Atom properly. [linter-bootlint](https://github.com/AtomLinter/linter-bootlint/blob/master/circle.yml) again is a good example since I just set that up a week or so ago :wink:.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+os:
+  - linux
+  - osx
+
+before_install:
+  - if grep -qe '^darwin'<<<$OSTYPE; then brew update && brew install scala; fi
+
+env:
+  global:
```
Comment (Arcanemagus) [inline .travis.yml]: We don't need this section unless we are utilizing it. It's for pre-installing packages the one you are testing depends on, but the package should be doing this on its own with `atom-package-deps`.

File: appveyor.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  - mkdir -p %LOCALAPPDATA%\scala
+  - 7z x scala-2.11.7.zip -o%LOCALAPPDATA%\scala
+  - SET PATH=%LOCALAPPDATA%\scala\scala-2.11.7\bin;%LOCALAPPDATA%\atom\bin;%PATH%
+  - appveyor DownloadFile https://atom.io/download/windows -FileName AtomSetup.exe
+  - AtomSetup.exe /silent
+
+build_script:
+  - cd %APPVEYOR_BUILD_FOLDER%
+  - SET PATH=%LOCALAPPDATA%\java\bin;%LOCALAPPDATA%\scala\bin;%LOCALAPPDATA%\atom\bin;%PATH%
+  - atom -v
```
Comment (Arcanemagus) [inline appveyor.yml]: Mind trying `atom --version` here? `atom -v` is broken for some reason even though it works locally, maybe `atom --version` will work.

**Comment by andystanton** [comment]
@Arcanemagus I've made those changes and squashed into a single commit, and all the builds pass on my fork. If there are no more comments I'll add the build jobs for the main repo and then merge the PR.

**Comment by Arcanemagus** [comment]
LGTM, merge away!

**Pull request merged**